### PR TITLE
FIX - Default Workstation on bom

### PR DESCRIPTION
**Pull request by atm-lucasmantegari**
# FIX *Default workstation on bom*

In the services of a BOM, when adding a line, dynamic management of workstations by default depending on the selected service. When modifying, possibility of emptying the default workstation field.

### CODE REVIEW

**Comment by atm-lucasmantegari** [comment]
@eldy  I don't understand the errors in phpstan and pre-commit, could you give more details.
I have the impression that the errors come from files not changed by me.

**Comment by eldy** [comment]
> @eldy I don't understand the errors in phpstan and pre-commit, could you give more details. I have the impression that the errors come from files not changed by me.

Yes, when you see error comes from another file than yours, you can repull and repush or just do nothing. This is solved by myself regularly by using the pull/push button of github called "Update branch"...

File: htdocs/bom/class/bom.class.php [hunk 1]
```diff
@@ -792,7 +792,7 @@ public function updateLine($rowid, $qty, $qty_frozen = 0, $disable_stock_change
 					$line->array_options[$key] = $array_options[$key];
 				}
 			}
-			if ($fk_default_workstation > 0 && $line->fk_default_workstation != $fk_default_workstation) {
+			if ($line->fk_default_workstation != $fk_default_workstation) {
```
Comment (eldy) [inline htdocs/bom/class/bom.class.php:795]: Why removing the test
`$fk_default_workstation > 0 &&`

Comment (atm-lucasmantegari) [reply htdocs/bom/class/bom.class.php:795]: I deleted this because it prevented emptying the field when you did not want to apply the default workstation

**Pull request merged**